lib/cli/ui/prompt/interactive_options.rb

```diff
@@ -141,7 +163,36 @@ def select_bool(char)
           @redraw = true
         end
 
+        def build_selection(char)
+          @active = (@active.to_s + char).to_i
+          @redraw = true
+        end
+
+        def chop_selection
+          @active = @active.to_s.chop.to_i
+          @redraw = true
+        end
+
+        def update_search(char)
+          @redraw = true
+
+          # Control+D or Backspace on empty search closes search
```

Comment:
Can you show this information to the user when filtering?

Reply:
Done.  I've also added a similar message when selecting a line that explains how to exit that mode.